### PR TITLE
OnBoarding Android fixes & Sidebar Screen

### PR DESCRIPTION
**Pull request by edencakir**
- Created Sidebar shared UI components (Header, title, item)
- Progressive Image Component children enabled (only with overlay)
- Sidebar Screen and other inner screens created.
- Login and Signup form icons horizontally centered.
- OnBoarding screen is now scrollable - Android only.
- Fixed: Facebook login button (footer) animating above the keyboard on Android.

### CODE REVIEW

**Comment by JoshFabian** [comment]
@blargity or @BilalBudhani mind reviewing this?

**Comment by BilalBudhani** [comment]
@JoshFabian @blargity looks to me. @edencakir Can you please post screen shots of the UI you worked on this PR. It gets easier to compare and review.

**Comment by edencakir** [comment]
@BilalBudhani https://projects.invisionapp.com/share/GBBRMN5UT#/screens/247954633

File: src/components/Forms/tcomb/textbox.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function textbox(locals) {
           borderBottomWidth: 0.5,
         }}
       >
-        <View style={{ width: 25 }}>
+        <View style={{ width: 25, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (joshmadewell) [inline src/components/Forms/tcomb/textbox.js]: prefer all styles to be using a StyleSheet. This is a `View` so it should play well with `StyleSheet`s, any time you use a component from `native-base` you'll have to do `StyleSheet.flatten(styles.styleName)`

File: src/routes/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { StackNavigator } from 'react-navigation';
+import navigationOptions from './navigationOptions';
+import SidebarScreen from '../screens/Sidebar/SidebarScreen';
```
Comment (joshmadewell) [inline src/routes/sidebar.js]: if you merge in the latest master we have an alias for the src folder now and you can change all your relative paths to start with `kitsu`

Good rule of thumb: if its' not in your current folder, start with `kitsu`

File: src/screens/Sidebar/Blocking.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      { tintColor },
+    ) => (
+      <Image
+        source={menu}
+        style={{ tintColor, width: 20, height: 21 }}
+      />
+    ),
+  });
+
+  onUnblockUser = (user) => {
```
Comment (joshmadewell) [inline src/screens/Sidebar/Blocking.js]: should mark this with a TODO or remove it

File: src/screens/Sidebar/Blocking.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    borderBottomWidth: 0,
+    marginLeft: 0 // FUCKING STUPID NATIVEBASE.
+  },
+};
+  
+
+const mapStateToProps = ({ user }) => {
```
Comment (joshmadewell) [inline src/screens/Sidebar/Blocking.js]: is it necessary to event connect this component to redux?

File: src/screens/Sidebar/ImportLibrary.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    paddingHorizontal: 8,
+    paddingVertical: 10,
+    borderRadius: 0,
+    borderColor: 'white',
+    flex: 1,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    borderBottomWidth: 0,
+    marginLeft: 0 // FUCKING STUPID NATIVEBASE.
```
Comment (joshmadewell) [inline src/screens/Sidebar/ImportLibrary.js]: you could just _not_ use native base :) but I agree

File: src/screens/Sidebar/common/SidebarListItem.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    flex: 1,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    borderBottomWidth: 0,
+    marginLeft: 0 // FUCKING STUPID NATIVEBASE.
+  },
+};
+
+export default SidebarListItem;
```
Comment (joshmadewell) [inline src/screens/Sidebar/common/SidebarListItem.js]: add new line

File: src/screens/Sidebar/common/SidebarTitle.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+SidebarTitle.propTypes = {
+  title: PropTypes.string.isRequired,
+};
+
+SidebarTitle.defaultProps = {
+  title: 'Settings',
+};
+
+export default SidebarTitle;
```
Comment (joshmadewell) [inline src/screens/Sidebar/common/SidebarTitle.js]: add new line

File: src/screens/Sidebar/Blocking.js (hunk 4 above)
Comment (edencakir) [inline src/screens/Sidebar/Blocking.js]: I haven't implemented the logic side of these views. I'll definitely consider connecting for each

**Comment by edencakir** [comment]
@joshmadewell Thanks for the feedback. Done editing

**Comment by joshmadewell** [comment]
I think this looks good for now -- in general -- I would try to put _all_ styles in a stylesheet -- but I'm not sold on whether that should be a hard rule or not

I think let someone else who's _more_ familiar with this code base take a look before merging

**Comment by edencakir** [comment]
@joshmadewell the reason I'm skipping them, for now, is #14. Instead of flattening, creating may be a better option -- not sure of the performance tho. But I'll definitely separate style definitions into a separate file.

After taking a look at #14 can you decide whether I should export the styles now or wait for it?

**Comment by joshmadewell** [comment]
aha, that makes sense -- the work will have to get done either way so it's either now or then :D up to you on this one :D

**Comment by edencakir** [comment]
@JoshFabian I should better do them later when we update our codebase. There are many plain style objects in other files, so I'll be taking care of them at one sitting 🕺

**Pull request merged**